pennylane_lightning/lightning_qubit.py

```diff
@@ -263,22 +273,35 @@ def reset(self):
     @property
     def state(self):
         # Flattening the state.
-        shape = (1 << self.num_wires,)
+        dim = 1 << self.num_wires
+        batch_size = self._get_batch_size(self._pre_rotated_state, (2,) * self.num_wires, dim)
+        # Do not flatten the state completely but leave the broadcasting dimension if there is one
+        shape = (batch_size, dim) if batch_size is not None else (dim,)
         return self._reshape(self._pre_rotated_state, shape)
 
     def _apply_state_vector(self, state, device_wires):
         """Initialize the internal state vector in a specified state.
+
         Args:
             state (array[complex]): normalized input state of length ``2**len(wires)``
-                or broadcasted state of shape ``(batch_size, 2**len(wires))``
             device_wires (Wires): wires that get initialized in the state
         """
 
         # translate to wire labels used by device
         device_wires = self.map_wires(device_wires)
 
         state = self._asarray(state, dtype=self.C_DTYPE)
+
+        dim = 1 << len(device_wires)
+        batch_size = self._get_batch_size(state, (dim,), dim)
+
         output_shape = [2] * self.num_wires
+        if batch_size is not None:
+            output_shape.insert(0, batch_size)
+
+        # Lightning doesn't support broadcasted state vector initialization.
+        if batch_size:
+            raise ValueError("Lightning doesn't support broadcasted state vector initialization")
```

Comment:
Should we move this up, before the creation of `output_shape`? Does not make a true difference, but conceptually it makes more sense not to create `output_shape` for the case `batch_size is not None`. Also, we can skip the check `if batch_size is not None` in that case (which is effectively done twice currently).

Reply:
Nice suggestion. Thank you.